### PR TITLE
Foundations for CI

### PR DESCRIPTION
**Pull request by inverse**
Laying the way for the foundations of setting up some automation. I am not sure how best to do this on a forked repo but verified it on my fork.

See: https://github.com/inverse/tell-me-your-secrets/pull/1/checks?check_run_id=861563069

In this PR

- Add foundations for CI that runs a very basic test suite on PR
- Minor style changes (we could check this by using a linter but didnt wanna go overboard)

p.s. do you squash when you merge? It makes history look cleaner - you can set this as the default in the settings.

### CODE REVIEW

File: tell_me_your_secrets/__main__.py [hunk 1]
```diff
@@ -53,47 +53,47 @@
...
+    def __init__(self, part: str, name: str, signature: str):
+        super().__init__(part, name, signature)
         self.regex = re.compile(self.signature)
           
-    def match(self, file_path,file_content) -> bool:
+    def match(self, file_path: str, file_content: str) -> bool:
         compare_variable = None
         if self.part == 'extension':
-            compare_variable = find_extension(file_path)
+            compare_variable = find_extension(file_path)[1:]
```
Comment (inverse) [inline tell_me_your_secrets/__main__.py:78]: This was actually returning like `.txt` when we the base rules never define leading period

Comment (valayDave) [reply tell_me_your_secrets/__main__.py:78]: Fascinating. Is it happening for all extensions?

Comment (inverse) [reply tell_me_your_secrets/__main__.py:78]: You can try it yourself

```
malachi@pulsar-work ~ $ python
Python 3.8.3 (default, May 17 2020, 18:15:42) 
[GCC 10.1.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pathlib
>>> pathlib.Path('/home/malachi/Downloads/output.log').suffix
'.log'
```

We could add additional test cases around https://github.com/valayDave/tell-me-your-secrets/pull/8/files#diff-e5d9203764fd4dc92a48e417f9f961bdR20

But would require some IO mocking I suspect

**Comment by inverse** [comment]
@valayDave anything else you would add to this right now? The coverage is around 40% with this change - which once we add #10 we can get more insights easier onto the coverage.

**Comment by valayDave** [comment]
@inverse : In what order should I merge the branches?

**Comment by inverse** [comment]
@valayDave I'd say start with this one... if there are conflicts I can fix them

**Comment by inverse** [comment]
Anything I can do to help with these PR's to unblock? Happy to contribute more on this project!

File: tell_me_your_secrets/__main__.py (hunk 1 above)
Comment (inverse) [reply tell_me_your_secrets/__main__.py:78]: @valayDave the alternative would be to suffix all the regex `extension` rules with `\.`

Comment (valayDave) [reply tell_me_your_secrets/__main__.py:78]: This should be fine.

**Comment by valayDave** [comment]
Hey, Can you also increment the version number of `setup.py`? I can directly integrate new build to PyPI after merging.

**Comment by valayDave** [comment]
I will get most of these branches merged this week.

**Pull request merged**